mule-extensions-api/src/test/java/org/mule/runtime/extension/api/describer/MuleArtifactDescriberBuilderTestCase.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.describer;
+
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import org.mule.runtime.api.meta.MuleVersion;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class MuleArtifactDescriberBuilderTestCase {
+
+  public static final String ARTIFACT_NAME = "artifact-name";
+  public static final MuleVersion MULE_VERSION = new MuleVersion("4.0.0");
+  public static final String DESCRIBER_ID = "ID-1";
+  private MuleArtifactDescriberBuilder builder;
+
+  @Before
+  public void before() {
+    builder = new MuleArtifactDescriberBuilder();
+  }
+
+  @Test
+  public void build() {
+    builder.setName(ARTIFACT_NAME).setMinMuleVersion(MULE_VERSION);
+    builder.withExtensionModelDescriber().setId(DESCRIBER_ID);
+    MuleArtifactDescriber muleArtifactDescriber = builder.build();
+
+    assertThat(muleArtifactDescriber.getName(), is(ARTIFACT_NAME));
+    assertThat(muleArtifactDescriber.getMinMuleVersion(), is(MULE_VERSION));
+    assertThat(muleArtifactDescriber.getExtensionModelDescriptor().getId(), is(DESCRIBER_ID));
+    assertThat(muleArtifactDescriber.getExtensionModelDescriptor().getAttributes().size(), is(0));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void blankName() {
+    builder.setName(null);
```

Comment:
Better to use a ExpectedException rule right before the exception is expected to happen